File: rerun-reports/lane-b/054/20260828-041803/results/identity/data-plane/python/auth/identity-dp-python-credential-chain/python-azure-skills/azure-skill-mcp-microsoft-skill/generated-code/environment_detector.py
"""Detect whether the process is running locally, in CI, or in production."""

from __future__ import annotations

import os
from enum import Enum
from typing import Mapping


class RuntimeEnvironment(str, Enum):
    DEV = "dev"
    CI = "ci"
    PRODUCTION = "production"


CI_MARKERS = (
    "CI",
    "GITHUB_ACTIONS",
    "GITLAB_CI",
    "TF_BUILD",
    "BUILD_BUILDID",
    "PIPELINE_WORKSPACE",
    "SYSTEM_TEAMFOUNDATIONCOLLECTIONURI",
    "JENKINS_URL",
)

PRODUCTION_MARKERS = (
    "IDENTITY_ENDPOINT",
    "MSI_ENDPOINT",
    "IMDS_ENDPOINT",
    "IDENTITY_HEADER",
    "MSI_SECRET",
    "WEBSITE_INSTANCE_ID",
    "CONTAINER_APP_NAME",
    "KUBERNETES_SERVICE_HOST",
    "AZURE_FEDERATED_TOKEN_FILE",
)


def detect_environment(
    environ: Mapping[str, str] | None = None,
) -> RuntimeEnvironment:
    """Classify the runtime using environment markers, with CI taking precedence."""
    values = os.environ if environ is None else environ

    if _has_marker(values, CI_MARKERS):
        return RuntimeEnvironment.CI
    if _has_marker(values, PRODUCTION_MARKERS):
        return RuntimeEnvironment.PRODUCTION
    return RuntimeEnvironment.DEV


def _has_marker(values: Mapping[str, str], names: tuple[str, ...]) -> bool:
    return any(values.get(name, "").strip() for name in names)

File: rerun-reports/lane-a/011/20260827-234017/results/identity/data-plane/java/auth/identity-dp-java-credential-chain/java-azure-skills/azure-skill-mcp/generated-code/src/main/java/com/example/azureidentity/EnvironmentDetector.java
package com.example.azureidentity;

import java.util.Map;
import java.util.stream.Stream;

public final class EnvironmentDetector {
    private static final String[] CI_MARKERS = {
        "CI",
        "TF_BUILD",
        "BUILD_BUILDID",
        "BUILD_SOURCESDIRECTORY",
        "SYSTEM_TEAMFOUNDATIONCOLLECTIONURI",
        "GITHUB_ACTIONS",
        "GITLAB_CI",
        "JENKINS_URL"
    };

    private static final String[] PRODUCTION_MARKERS = {
        "IDENTITY_ENDPOINT",
        "MSI_ENDPOINT",
        "IMDS_ENDPOINT",
        "AZURE_FEDERATED_TOKEN_FILE",
        "WEBSITE_INSTANCE_ID",
        "CONTAINER_APP_NAME",
        "KUBERNETES_SERVICE_HOST"
    };

    public DeploymentEnvironment detect() {
        return detect(System.getenv());
    }

    public DeploymentEnvironment detect(Map<String, String> environment) {
        if (hasAnyNonBlank(environment, CI_MARKERS)) {
            return DeploymentEnvironment.CI;
        }
        if (hasAnyNonBlank(environment, PRODUCTION_MARKERS)) {
            return DeploymentEnvironment.PRODUCTION;
        }
        return DeploymentEnvironment.DEV;
    }

    private boolean hasAnyNonBlank(Map<String, String> environment, String[] names) {
        return Stream.of(names).map(environment::get).anyMatch(EnvironmentDetector::isNonBlank);
    }

    private static boolean isNonBlank(String value) {
        return value != null && !value.isBlank();
    }
}
